lib/ohai/plugins/darwin/cpu.rb

```diff
@@ -22,10 +22,12 @@
 
   collect_data(:darwin) do
     cpu Mash.new
-    so = shell_out("sysctl -n hw.physicalcpu")
+    so = shell_out("sysctl -n hw.packages")
     cpu[:real] = so.stdout.to_i
-    so = shell_out("sysctl -n hw.logicalcpu")
+    so = shell_out("sysctl -n hw.physicalcpu")
     cpu[:total] = so.stdout.to_i
+    so = shell_out("sysctl -n hw.logicalcpu")
+    cpu[:cores] = so.stdout.to_i
```

Comment:
well, this is confusing.  at first glance it seems backwards from what i'd expect.

"cores" should be physical cores -- i would expect '4' here
"total" should be the logical cores things like the scheduler sees - i would expect '8' here
"real" should be how many packages i haven plugged into the motherboard - i would expect '1' here

i haven't looked at how we define these on linux or what the APIs return on darwin, but your cores and total read backwards from what i would expect.

(and definitely include some comments if things really need to be confusing).